tile_fetch.py

```diff
@@ -114,6 +125,23 @@ async def load_tiles(info, z=-1, outfile=None, quality=90):
         )
         z = len(info.tile_info) - 1
 
+    ## Ensuring image resolution fits in JPEG - two pass 
+    if info.tile_info[z].size[0] > 65535 or info.tile_info[z].size[1] > 65535:
+        print(
+            'Zoom level {r} too high for JPEG output, using next zoom level {next_z} instead'.format(
+                r=z,
+                next_z=z-1)
+        )
+        z = z-1
+        
+    if info.tile_info[z].size[0] > 65535 or info.tile_info[z].size[1] > 65535:
+        print(
+            'Zoom level {r} *still* too high for JPEG output, using next zoom level {next_z} instead'.format(
+                r=z,
+                next_z=z-1)
+        )
+        z = z-1
```

Comment:
If the user has specifically requested a zoom level, we should respect it (and save as PNG if the image is too large)

But when we are choosing the zoom level automatically, we can run this instead of always choosing `z = len(info.tile_info) - 1`.

And this should be done in a loop, not with 2 successive checks.

Reply:
We can save it as a PNG if the zoom is too large, but would then have to skip all code appending metadata to the file - PNG doesn't support it.

Correct - I'll make it into a loop.

Reply:
Yes, the code that appends metadata must be in a conditional.

Reply:
Should be done in latest commit... but please let me know if I missed something.